lib/sshutils/utils.go

```diff
@@ -21,9 +21,30 @@ package sshutils
 import (
 	"net"
 	"strconv"
+	"strings"
+
+	"github.com/gravitational/trace"
 )
 
 // JoinHostPort is a wrapper for net.JoinHostPort that takes a uint32 port..
 func JoinHostPort(host string, port uint32) string {
 	return net.JoinHostPort(host, strconv.Itoa(int(port)))
 }
+
+// SplitHostPort is a wrapper for net.SplitHostPort that returns a uint32 port.
+// Note that unlike net.SplitHostPort, a missing port is valid and will return
+// a zero port.
+func SplitHostPort(addr string) (string, uint32, error) {
```

Comment:
I've updated `SplitHostPort` to use `ParseHostPortAddr` under the hood. I kind of still want it to exist here in `lib/sshutils` because SSH messages always expect uint32 and this saves us from having to cast everywhere. Not a big deal if you still want to change it though.